### ISSUE TITLE
Doesn't work if the job is being built on a slave

### ISSUE THREAD

**Issue by urjitbhatia** [awslabs/aws-codedeploy-plugin#63]
If I run my build job on a jenkins slave, the plugin still looks for the appspec.yml on the master. Is there any workaround for this?

**Comment by Dries-Vandenneucker** [comment]
I don't think this is a bug. We only run jobs on slaves and they all use codedeploy. 

Is it possible you use the setting "Use own workspace" under General? "...If this path is relative, it's resolved against the "remote FS root" directory of the slave, or $JENKINS_HOME on the master." So maybe you are resolving against the master?

**Comment by delpinoconsulting** [comment]
Sorry to resurrect the thread, but I have the same issue. I'm building on a Jenkins node, but if I use Per-deployment appspec files, the job never finds it regardless if I specify the workspace. Would appreciate some configuration help.
